### ISSUE TITLE
New security groups are created even though Networking->SecurityGroups of HeadNode and all queues and EFS mount target has the same security group and the EFS is created beforehand

### ISSUE THREAD

**Issue by mopk** [aws/aws-parallelcluster#5232]
Hello,
I want' to avoid creation of any new security groups. The ones created by Parallel Cluster v3.5.0 I currently use are like all allowed within them and thus duplicating my existing security group I want to use.

My config is:
```
Region: <my-region>
Image:
  CustomAmi: <my-ami-id>
  Os: <my-ami-os>
SharedStorage:
  - MountDir: <my-mount-dir-path>
    Name: my-efs-mount
    StorageType: Efs
    EfsSettings:
      FileSystemId: <my-efs-file-system-id>
HeadNode:
  Imds:
    Secured: false
  InstanceType: <my-instance-type>
  Networking:
    SubnetId: <my-subnet-id>
    SecurityGroups:
      - <my-security-group-id>
  Ssh:
    KeyName: <my-key>
  LocalStorage:
    RootVolume:
      Size: 50
Scheduling:
  Scheduler: slurm
  SlurmSettings:
    ScaledownIdletime: 5
  SlurmQueues:
    - Name: queue1
      ComputeResources:
        - Name: <compute-resource-name>
          InstanceType: <my-instance-type>
      Networking:
        SubnetIds:
          - <my-subnet-id>
        SecurityGroups:
          - <my-security-group-id>
      ComputeSettings:
        LocalStorage:
          RootVolume:
            Size: 50
```

My EFS file system is created beforehand and its mount target has the same subnet and security group as the HeadNode and queue.
(They are all in the same VPC, region, availability zone.)

NOTE: The problem doesn't happen with an FSx Lustre file system created beforehand.

Is there any explicit flag/option to make pcluster CLI avoid creation of any new SGs?
I expect that creation is to be avoided when I set my existing SG to Head Node and all the queues as per documentation.
But by some unclear to me reason this doesn't happen as expected. For now the behaviour when one creates an EFS beforehand and use the same groups for its mount target, HeadNode and all queues and pcluster creates new security groups looks like a bug.

**Comment by hanwen-pcluste** [comment]
Hi mopk,

If you provide security groups to `SecurityGroups` under `HeadNode` and all `SlurmQueues`, and use shared storage created beforehand, ParallelCluster does not create any security groups.
If you ask ParallelCluster to create some FSx or EFS, ParallelCluster has to create some security groups and there is no CLI flag/option to disable it.

Can you tell us why you would avoid creating any security groups? Is it because of security reasons?

Thank you,
Hanwen

**Comment by mopk** [comment]
> Hi mopk,
> 
> If you provide security groups to `SecurityGroups` under `HeadNode` and all `SlurmQueues`, and use shared storage created beforehand, ParallelCluster does not create any security groups. If you ask ParallelCluster to create some FSx or EFS, ParallelCluster has to create some security groups and there is no CLI flag/option to disable it.
> 
> Can you tell us why you would avoid creating any security groups? Is it because of security reasons?
> 
> Thank you, Hanwen

I'm afraid your suggestion doesn't work in the case described by me above. Have you read the config posted by me and my comments around it?
That's all what the issue is about. The suggestion that is to:

> provide security groups to SecurityGroups under HeadNode and all SlurmQueues, and use shared storage created beforehand, ParallelCluster does not create any security groups.

doesn't work in the case of an EFS filesystem created beforehand. The config posted explicitly has the security groups set in the places you mention (HeadNode->Networking->SecurityGroup, SlurmQueues->queue1->Networking->SecurityGroups). The EFS used is created beforehand and has the same VPC, subnet and security group as the HeadNode and the queue of the cluster.

I can easily reproduce the issue and have done it several times during last few days in slightly various configurations while trying to solve it.
The NOTE about FSx Lustre makes me upset as the issue is not the case if trying to mount an FSx Lustre created beforehand making the behaviour inconsistent depending on the type of a mount.
In that case no SecurityGroups are created as per your suggestion. By the way, which part of the documentation does the suggestion come from? (I've read about it before but never seen it explicitly mentioned in the docs while one might deduce that thoroughly reading HeadNode and Scheduling sections of Cluster configuration reference.)

Answering you question on why I want to avoid security groups I can tell that in practice in many companies the guys responsible for cyber/network security are those responsible for creation of security groups as well thus making it virtually impossible to create security groups on the fly not passing security audit first. That's my case as well.

**Comment by hanwen-pcluste** [comment]
Hi mopk,

We have identified the issue: security groups are created by ParallelCluster when external EFS are provided. The security groups are not attached anywhere. This is a problem when users don't want to create any security groups because of their account security policies.

Bug location: https://github.com/aws/aws-parallelcluster/blob/develop/cli/src/pcluster/templates/cluster_stack.py#L811

We will discuss the issue within our team next week.

Thank you for helping us identifying the issue!
Hanwen

**Comment by hanwen-pcluste** [comment]
https://github.com/aws/aws-parallelcluster/pull/5246 has been merged. This bug will be fixed in the next release. 
Thank you for reporting this bug!

**Comment by enrico-usai** [comment]
Issue fixed on 3.6.0 release: https://github.com/aws/aws-parallelcluster/releases/tag/v3.6.0
